roxie/ccd/ccdcontext.cpp

```diff
@@ -3497,11 +3497,11 @@ class CRoxieServerContext : public CRoxieContextBase, implements IRoxieServerCon
     {
         CriticalBlock b(contextCrit);
         StringBuffer expandedName;
-        expandLogicalFilename(expandedName, fileName, workUnit, false, !workUnit);
+        expandLogicalFilename(expandedName, fileName, workUnit, false, !serverQueryFactory->isDynamic());
```

Comment:
I'm not sure you would be able to move the files to the local dali (at deploy time) if dynamic was specified, since they won't be mentioned in the workunit. That suggests to me that dynamic files need to resolve foreign at runtime, even in deployed queries. Hopefully the use of dynamic in a deployed query is unusual!